presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -583,6 +621,11 @@ public InternalResourceGroup getOrCreateSubGroup(String name)
         }
     }
 
+    private boolean isSubResourceGroupStatic(Optional<ResourceGroupSelectionContext> resourceGroupSelectionContext)
+    {
+        return !resourceGroupSelectionContext.equals(Optional.of(createResourceGroupSelectionContext(id.getSegments().size())));
```

Comment:
Coudn't understand the logic here, how are we deciding if a subgroup is static vs dynamic.  can you explain the logic here?

Reply:
resourceGroupSelectionContext is holding the dynamic segment index. 
For example:
abc_static.xyx_static.**qwe_dynamic**.sdq_static.
dynamic segment index is 2.
id.segment.size() is giving the size of the parent. This is also the index of the subgroup. so if parent.segment.size == dynamic segment index than the sub groups is dynamic.